File: package/android/src/main/java/com/margelo/nitro/nitromaps/Region+Validity.kt
package com.margelo.nitro.nitromaps

/** A `NaN` or non-positive span puts the southern edge above the northern one, which `LatLngBounds` rejects. */
internal fun Region.isValid(): Boolean =
  isValidCoordinate(latitude, longitude) &&
    latitudeDelta.isFinite() &&
    latitudeDelta > 0.0 &&
    longitudeDelta.isFinite() &&
    longitudeDelta > 0.0

File: package/ios/Region+Validity.swift
import CoreLocation

extension Region {
  var isValid: Bool {
    CLLocationCoordinate2DIsValid(
      CLLocationCoordinate2D(latitude: latitude, longitude: longitude)
    ) && RegionSpan.isDrawable(latitudeDelta: latitudeDelta, longitudeDelta: longitudeDelta)
  }
}
